org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/compiler/OnTheFlyJavaCompiler2.java

```diff
@@ -0,0 +1,187 @@
+				richString.append(
+						Joiner.on("\n").join(IterableExtensions.filter(result.getCompilationProblems(), CategorizedProblem::isError)));
+				richString.newLineIfNotEmpty();
+				richString.newLine();
+				richString.append("Code was:");
+				richString.newLine();
+				richString.append("=========");
+				richString.newLine();
+				richString.append(Joiner.on("\n=========\n").join(sources.values()));
+				richString.newLineIfNotEmpty();
+				richString.append("=========");
+				richString.newLine();
+				throw new IllegalArgumentException(richString.toString());
+			}
+			ClassLoader classLoader = result.getClassLoader();
+			return IterableExtensions.toMap(Iterables.transform(sources.keySet(), (String it) -> {
+				try {
+					return classLoader.loadClass(it);
+				} catch (ClassNotFoundException e) {
+					throw Exceptions.sneakyThrow(e);
+				}
+			}), Class::getName);
+		} catch (Throwable t) {
+			if (t instanceof ClassNotFoundException) {
+				ClassNotFoundException e = (ClassNotFoundException) t;
+				StringConcatenation richString = new StringConcatenation();
+				richString.append(e.getMessage());
+				richString.append(" ");
+				richString.newLineIfNotEmpty();
+				richString.append("source :");
+				richString.newLine();
+				richString.append("\t");
+				richString.append(sources, "\t");
+				richString.newLineIfNotEmpty();
+				richString.newLine();
+				richString.append("PROBLEMS : ");
+				richString.newLine();
+				richString.append("\t");
+				String _join_2 = Joiner.on("\n").join(result.getCompilationProblems());
+				richString.append(_join_2, "\t");
+				richString.newLineIfNotEmpty();
+				throw new IllegalStateException(richString.toString(), e);
+			} else {
+				throw Exceptions.sneakyThrow(t);
+			}
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	public <RT extends Object> Function0<RT> createFunction(String expression, Class<RT> returnType) {
+		try {
+			StringConcatenation richString = new StringConcatenation();
+			richString.append("public class __Generated implements org.eclipse.xtext.xbase.lib.Functions.Function0<");
+			richString.append(returnType.getName());
+			richString.append("> {");
+			richString.newLineIfNotEmpty();
+			richString.append("\t");
+			richString.append("public ");
+			richString.append(returnType.getName(), "\t");
+			richString.append(" apply() {");
+			richString.newLineIfNotEmpty();
+			richString.append("\t\t");
+			richString.append(expression, "\t\t");
+			richString.newLineIfNotEmpty();
+			richString.append("\t");
+			richString.append("}");
+			richString.newLine();
+			richString.append("}");
+			richString.newLine();
+			Class<?> clazz = compileToClass("__Generated", richString.toString());
+			return (Function0<RT>) clazz.getDeclaredConstructor().newInstance();
+		} catch (InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException
+				| NoSuchMethodException | SecurityException e) {
+			throw Exceptions.sneakyThrow(e);
```

Comment:
throw new RuntimeException

Reply:
did not want to change any semantics on a deprecated class